pkg/util/proto/testing/gvk.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"github.com/googleapis/gnostic/OpenAPIv2"
+	"k8s.io/kube-openapi/pkg/util/proto"
+)
+
+// groupVersionKindExtensionKey is the key used to lookup the
```

Comment:
We should not have groupVersionKindExtensionKey here. The extension is kubernetes specific. I suggest you make it configurable somehow and set it in the importer of this code.

Reply:
Never mind, This is in test code. It is fine. nit: I would name it something like testExtensionKey and add it to the comment that only for testing we use some kubernetes spec that has this key.